From 0035a4704557ba66824c08d5759d6e743747410b Mon Sep 17 00:00:00 2001
From: Jan Glauber <jglauber@digitalocean.com>
Date: Wed, 17 Mar 2021 15:04:39 +0100
Subject: [PATCH 2571/4917] md: Fix missing unused status line of /proc/mdstat
Patch-mainline: 5.10.50
References: bsc#0000000
Git-commit: 0035a4704557ba66824c08d5759d6e743747410b

commit 7abfabaf5f805f5171d133ce6af9b65ab766e76a upstream.

Reading /proc/mdstat with a read buffer size that would not
fit the unused status line in the first read will skip this
line from the output.

So 'dd if=/proc/mdstat bs=64 2>/dev/null' will not print something
Like: unused devices: <none>

Don't return NULL immediately in start() for v=2 but call
show() once to print the status line also for multiple reads.

Cc: stable@vger.kernel.org
Fixes: 1f4aace60b0e ("fs/seq_file.c: simplify seq_file iteration code and interface")
Signed-off-by: Jan Glauber <jglauber@digitalocean.com>
Signed-off-by: Song Liu <songliubraving@fb.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

---
 drivers/md/md.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/md/md.c b/drivers/md/md.c
index 8ab8afe5546f..288d26013de2 100644
--- a/drivers/md/md.c
+++ b/drivers/md/md.c
@@ -8187,7 +8187,11 @@ static void *md_seq_start(struct seq_file *seq, loff_t *pos)
 	loff_t l = *pos;
 	struct mddev *mddev;
 
-	if (l >= 0x10000)
+	if (l == 0x10000) {
+		++*pos;
+		return (void *)2;
+	}
+	if (l > 0x10000)
 		return NULL;
 	if (!l--)
 		/* header */
-- 
2.32.0

